calculator_test.go

```diff
@@ -2,23 +2,195 @@ package calculator_test
 
 import (
 	"calculator"
+	"math/rand"
 	"testing"
+	"time"
 )
 
+// GenerateNums it is a helper function
+// that produces a slice of two float numbers.
+func GenerateNums() []float64 {
+	// Make generator to produce different
+	// numbers each time it runs (make it nondeterministic).
+	s := rand.NewSource(time.Now().UnixNano())
+	r := rand.New(s)
+
+	a := r.Float64() * float64(r.Intn(10000))
+	b := r.Float64() * float64(r.Intn(10000))
+	return []float64{a, b}
+}
+
 func TestAdd(t *testing.T) {
 	t.Parallel()
-	var want float64 = 4
-	got := calculator.Add(2, 2)
-	if want != got {
-		t.Errorf("want %f, got %f", want, got)
+
+	tt := []struct {
+		name        string
+		nums        []float64
+		want        float64
+		expectedErr bool
+	}{
+		{name: "Add two positive numbers", nums: []float64{1, 2}, want: 3, expectedErr: false},
+		{name: "Add two negative numbers", nums: []float64{-4.5, -5.5}, want: -10, expectedErr: false},
+		{name: "Add multiple numbers", nums: []float64{-4.5, -5.5, 10, 2.5}, want: 2.5, expectedErr: false},
+		{name: "Add no numbers", nums: []float64{}, want: 0, expectedErr: true},
+		{name: "Add one number", nums: []float64{10}, want: 10, expectedErr: true},
+	}
+
+	for _, tc := range tt {
+		got, err := calculator.Add(tc.nums...)
+
+		if err != nil && !tc.expectedErr {
```

Comment:
There's one possibility that we didn't catch here: what if the function _was_ supposed to return an error, but didn't?